packages/libs/error-stack/src/fmt/hook.rs

```diff
@@ -471,14 +483,14 @@ type BoxedFallbackHook =
 pub(crate) struct Hooks {
     // TODO: Remove `Option` when const `BTreeMap::new` is stabilized
     pub(crate) inner: Option<BTreeMap<TypeId, BoxedHook>>,
-    pub(crate) fallback: Option<BoxedFallbackHook>,
+    pub(crate) fallback: Option<BoxedHook>,
```

Comment:
I wonder if we need `fallback` at all. Technically, we don't as it's the same as `inner` but for `Frame` (which is a type we own).

Reply:
Yes, but(!) fallback has different semantics, it's only called when no hook was found, with the other PR (on phone will add reference later) this would mean that a non fallback &Frame hook would always execute.

I also believe that we cannot request_ref Frame or downcast_ref, but I might be wrong on that one.

Reply:
> I also believe that we cannot request_ref Frame or downcast_ref

I guess that's not possible, yes (and if it's possible we shouldn't allow this)

I guess that's the old memory vs performance discussion. As this is only stored once, it's probably the easiest to just keep it as it is.

No need to link the other PR, I think it's linked in several places here already and I'd resolve this comment anyway. 🚀